Sources/Purchasing/StoreKit2/StoreKit2Receipt.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2Receipt.swift
+//
+//  Created by MarkVillacampa on 26/10/23.
+
+import StoreKit
+
+/// A type that resembles the structure of a StoreKit 1 receipt using StoreKit 2 data.
+struct StoreKit2Receipt: Equatable {
+
+    struct SubscriptionStatus: Equatable {
+        /// Subscription Group Identifiers.
+        let subscriptionGroupId: String
+        /// JWS tokens of the renewal information.
+        let renewalInfoJWSTokens: [String]
+    }
+
+    /// The server environment where the receipt was generated.
+    let environment: StoreEnvironment
+
+    /// The current subscription status for each subscription group, including the renewal information.
+    let subscriptionStatus: [SubscriptionStatus]
+
+    /// The list of transaction JWS tokens purchased by the customer.
+    let transactions: [String]
+
+    /// The bundle identifier of the app.
+    let bundleId: String
+
+    /// The app version that the user originally purchased from the App Store.
+    let originalApplicationVersion: String?
+
+    /// The date the user originally purchased the app from the App Store.
+    let originalPurchaseDate: Date?
+
+}
+
+// MARK: -
+
+extension StoreKit2Receipt.SubscriptionStatus: Codable {}
+
+extension StoreKit2Receipt: Codable {
+
+    private enum CodingKeys: String, CodingKey {
+        case environment
+        case subscriptionStatus = "subscription_status"
+        case transactions
+        case bundleId = "bundle_id"
+        case originalApplicationVersion = "original_application_version"
+        case originalPurchaseDate = "original_purchase_date"
+    }
+
+    func encode(to encoder: Encoder) throws {
```

Comment:
Can you explain why you had to implement this manually?

Reply:
After introducing the `SubscriptionStatus` struct so it's more obvious what each field represents, I needed to customize the encoding so the JSON format for `subscription_status` fits what the API will expect (it's also less verbose):

```
"subscription_status" : {
  "{SUBSCRIPTION_GROUP_ID}": ["jws_token"]
}
```

instead of 

```
"subscription_status" : [{
   "subscriptionGroupId": "foo",
  "renewalInfoJWSTokens": ["bar"]
}]
```

Reply:
There is but it's not trivial to implement.

Let's at least add test coverage for this using snapshot tests.

Reply:
I just realized that this class is `Codable`, you've changed how it's _encoded_, but now how it's _decoded_, so doing:
```swift
expect(try receipt.encodeAndDecode()) == receipt
```

Would fail. I think you can change this to conform to `Encodable` only instead?